### PR TITLE
Fix PushEvent NullPointerException jenkinsci/github-plugin

### LINKED ISSUE #27202: jenkinsci/github-plugin Received PushEvent for null from ip/domain  NullPointerException DefaultPushGHEventSubscriber.onEvent
### Feature Description

main error happening on jenkins server PushEvent triggering

```
2023-09-23 04:19:37.733+0000 [id=17]	INFO	o.j.p.g.w.s.DefaultPushGHEventSubscriber#onEvent: Received PushEvent for null from  http://jenkinsdomain.com/github-webhook/
2023-09-23 04:19:37.733+0000 [id=17]	SEVERE	o.j.p.g.e.GHEventsSubscriber$4#applyNullSafe: Subscriber org.jenkinsci.plugins.github.webhook.subscriber.DefaultPushGHEventSubscriber failed to process SCMEvent
```

```
hook, skipping...
java.lang.NullPointerException
	at org.jenkinsci.plugins.github.webhook.subscriber.DefaultPushGHEventSubscriber.onEvent(DefaultPushGHEventSubscriber.java:79)
```

I find out that in https://github.com/jenkinsci/github-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github/webhook/subscriber/DefaultPushGHEventSubscriber.java#L76
`URL repoUrl = push.getRepository().getUrl();`
https://github.com/hub4j/github-api/blob/main/src/main/java/org/kohsuke/github/GHObject.java#L98-L100
```
public URL getUrl() {
        return GitHubClient.parseURL(url);
}
```
https://github.com/hub4j/github-api/blob/main/src/main/java/org/kohsuke/github/GHObject.java#L31
`private String url;`

it is trying to get **url** data from payload object of `GHRepository`

https://github.com/go-gitea/gitea/blob/main/modules/structs/repo.go#L50-L110
but in **Repository** struct there don't have any **url** variable
I test it adding the variable and it's now not showing NullPointerException on jenkins server.

```
2023-09-23 04:21:20.539+0000 [id=16]	INFO	o.j.p.g.w.s.DefaultPushGHEventSubscriber#onEvent: Received PushEvent for http://giteadomain.com/user/reponame from http://jenkinsdomain.com/github-webhook/
2023-09-23 04:21:20.607+0000 [id=16]	INFO	o.j.p.g.w.s.DefaultPushGHEventSubscriber$1#run: Poked test
2023-09-23 04:21:29.772+0000 [id=616]	INFO	c.c.jenkins.GitHubPushTrigger$1#run: SCM changes detected in test. Triggering #11
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Nabapadma-sarker**
Fixes #27202

### CODE REVIEW

File: services/convert/repository.go [hunk 1]
```diff
@@ -181,6 +181,7 @@ func innerToRepo(ctx context.Context, repo *repo_model.Repository, permissionInR
 		Parent:                        parent,
 		Mirror:                        repo.IsMirror,
 		HTMLURL:                       repo.HTMLURL(),
+		URL:                           repo.URL(),
```
Comment (CaiCandong) [inline services/convert/repository.go]: What's the difference between calling `repo.HTMLURL()` and `repo.URL()`?

Comment (Nabapadma-sarker) [reply services/convert/repository.go]: there is no difference. I made a new function for it if the repo.HTMLURL() function changed latter. remove it or keep?

Comment (yardenshoham) [reply services/convert/repository.go]: Just reuse `repo.HTMLURL`

Comment (Nabapadma-sarker) [reply services/convert/repository.go]: thanks for the suggestion, reused that.

**Comment by CaiCandong** [comment]
please run `make generate-swagger`  to fix it

**Comment by lunny** [comment]
Could we change jenkinsci side as `html_url`?

**Comment by Nabapadma-sarker** [comment]
> Could we change jenkinsci side to use `html_url`, it's a duplication.

https://docs.github.com/en/enterprise-server@3.6/webhooks/webhook-events-and-payloads

```
"repository": {
    "id": 17273051,
    "node_id": "MDEwOlJlcG9zaXRvcnkxNzI3MzA1MQ==",
    "name": "octo-repo",
    "full_name": "octo-org/octo-repo",
    "private": true,
    "owner": {
      "login": "octo-org",
      "id": 6811672,
      "node_id": "MDEyOk9yZ2FuaXphdGlvbjY4MTE2NzI=",
      "avatar_url": "https://avatars.githubusercontent.com/u/6811672?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/octo-org",
      "html_url": "https://github.com/octo-org",
      "followers_url": "https://api.github.com/users/octo-org/followers",
      "following_url": "https://api.github.com/users/octo-org/following{/other_user}",
      "gists_url": "https://api.github.com/users/octo-org/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/octo-org/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/octo-org/subscriptions",
      "organizations_url": "https://api.github.com/users/octo-org/orgs",
      "repos_url": "https://api.github.com/users/octo-org/repos",
      "events_url": "https://api.github.com/users/octo-org/events{/privacy}",
      "received_events_url": "https://api.github.com/users/octo-org/received_events",
      "type": "Organization",
      "site_admin": false
    },
    "html_url": "https://github.com/octo-org/octo-repo",
    "description": "My first repo on GitHub!",
    "fork": false,
    "url": "https://api.github.com/repos/octo-org/octo-repo",
    "forks_url": "https://api.github.com/repos/octo-org/octo-repo/forks",
    "keys_url": "https://api.github.com/repos/octo-org/octo-repo/keys{/key_id}",
```
here you can see github weebhook payload also using url, html_url for different context. I don't know what would be right url format though it's now resolved the null exception.

**Comment by lunny** [comment]
> > Could we change jenkinsci side to use `html_url`, it's a duplication.
> 
> https://docs.github.com/en/enterprise-server@3.6/webhooks/webhook-events-and-payloads
> 
> ```
> "repository": {
>     "id": 17273051,
>     "node_id": "MDEwOlJlcG9zaXRvcnkxNzI3MzA1MQ==",
>     "name": "octo-repo",
>     "full_name": "octo-org/octo-repo",
>     "private": true,
>     "owner": {
>       "login": "octo-org",
>       "id": 6811672,
>       "node_id": "MDEyOk9yZ2FuaXphdGlvbjY4MTE2NzI=",
>       "avatar_url": "https://avatars.githubusercontent.com/u/6811672?v=4",
>       "gravatar_id": "",
>       "url": "https://api.github.com/users/octo-org",
>       "html_url": "https://github.com/octo-org",
>       "followers_url": "https://api.github.com/users/octo-org/followers",
>       "following_url": "https://api.github.com/users/octo-org/following{/other_user}",
>       "gists_url": "https://api.github.com/users/octo-org/gists{/gist_id}",
>       "starred_url": "https://api.github.com/users/octo-org/starred{/owner}{/repo}",
>       "subscriptions_url": "https://api.github.com/users/octo-org/subscriptions",
>       "organizations_url": "https://api.github.com/users/octo-org/orgs",
>       "repos_url": "https://api.github.com/users/octo-org/repos",
>       "events_url": "https://api.github.com/users/octo-org/events{/privacy}",
>       "received_events_url": "https://api.github.com/users/octo-org/received_events",
>       "type": "Organization",
>       "site_admin": false
>     },
>     "html_url": "https://github.com/octo-org/octo-repo",
>     "description": "My first repo on GitHub!",
>     "fork": false,
>     "url": "https://api.github.com/repos/octo-org/octo-repo",
>     "forks_url": "https://api.github.com/repos/octo-org/octo-repo/forks",
>     "keys_url": "https://api.github.com/repos/octo-org/octo-repo/keys{/key_id}",
> ```
> 
> here you can see github weebhook payload also using url, html_url for different context. I don't know what would be right url format though it's now resolved the null exception.

So `url` means APIURL, it's not the same as `HTMLURL`.

**Comment by Nabapadma-sarker** [comment]
> > > Could we change jenkinsci side to use `html_url`, it's a duplication.
> > 
> > 
> > https://docs.github.com/en/enterprise-server@3.6/webhooks/webhook-events-and-payloads
> > ```
> > "repository": {
> >     "id": 17273051,
> >     "node_id": "MDEwOlJlcG9zaXRvcnkxNzI3MzA1MQ==",
> >     "name": "octo-repo",
> >     "full_name": "octo-org/octo-repo",
> >     "private": true,
> >     "owner": {
> >       "login": "octo-org",
> >       "id": 6811672,
> >       "node_id": "MDEyOk9yZ2FuaXphdGlvbjY4MTE2NzI=",
> >       "avatar_url": "https://avatars.githubusercontent.com/u/6811672?v=4",
> >       "gravatar_id": "",
> >       "url": "https://api.github.com/users/octo-org",
> >       "html_url": "https://github.com/octo-org",
> >       "followers_url": "https://api.github.com/users/octo-org/followers",
> >       "following_url": "https://api.github.com/users/octo-org/following{/other_user}",
> >       "gists_url": "https://api.github.com/users/octo-org/gists{/gist_id}",
> >       "starred_url": "https://api.github.com/users/octo-org/starred{/owner}{/repo}",
> >       "subscriptions_url": "https://api.github.com/users/octo-org/subscriptions",
> >       "organizations_url": "https://api.github.com/users/octo-org/orgs",
> >       "repos_url": "https://api.github.com/users/octo-org/repos",
> >       "events_url": "https://api.github.com/users/octo-org/events{/privacy}",
> >       "received_events_url": "https://api.github.com/users/octo-org/received_events",
> >       "type": "Organization",
> >       "site_admin": false
> >     },
> >     "html_url": "https://github.com/octo-org/octo-repo",
> >     "description": "My first repo on GitHub!",
> >     "fork": false,
> >     "url": "https://api.github.com/repos/octo-org/octo-repo",
> >     "forks_url": "https://api.github.com/repos/octo-org/octo-repo/forks",
> >     "keys_url": "https://api.github.com/repos/octo-org/octo-repo/keys{/key_id}",
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > here you can see github weebhook payload also using url, html_url for different context. I don't know what would be right url format though it's now resolved the null exception.
> 
> So `url` means APIURL, it's not the same as `HTMLURL`.

I build again using APIURL 
`URL:      repo.APIURL(),` 
that is also working with jenkins plugin. so do I need to change it to repo.APIURL() ?

**Comment by lunny** [comment]
> > > > Could we change jenkinsci side to use `html_url`, it's a duplication.
> > > 
> > > 
> > > https://docs.github.com/en/enterprise-server@3.6/webhooks/webhook-events-and-payloads
> > > ```
> > > "repository": {
> > >     "id": 17273051,
> > >     "node_id": "MDEwOlJlcG9zaXRvcnkxNzI3MzA1MQ==",
> > >     "name": "octo-repo",
> > >     "full_name": "octo-org/octo-repo",
> > >     "private": true,
> > >     "owner": {
> > >       "login": "octo-org",
> > >       "id": 6811672,
> > >       "node_id": "MDEyOk9yZ2FuaXphdGlvbjY4MTE2NzI=",
> > >       "avatar_url": "https://avatars.githubusercontent.com/u/6811672?v=4",
> > >       "gravatar_id": "",
> > >       "url": "https://api.github.com/users/octo-org",
> > >       "html_url": "https://github.com/octo-org",
> > >       "followers_url": "https://api.github.com/users/octo-org/followers",
> > >       "following_url": "https://api.github.com/users/octo-org/following{/other_user}",
> > >       "gists_url": "https://api.github.com/users/octo-org/gists{/gist_id}",
> > >       "starred_url": "https://api.github.com/users/octo-org/starred{/owner}{/repo}",
> > >       "subscriptions_url": "https://api.github.com/users/octo-org/subscriptions",
> > >       "organizations_url": "https://api.github.com/users/octo-org/orgs",
> > >       "repos_url": "https://api.github.com/users/octo-org/repos",
> > >       "events_url": "https://api.github.com/users/octo-org/events{/privacy}",
> > >       "received_events_url": "https://api.github.com/users/octo-org/received_events",
> > >       "type": "Organization",
> > >       "site_admin": false
> > >     },
> > >     "html_url": "https://github.com/octo-org/octo-repo",
> > >     "description": "My first repo on GitHub!",
> > >     "fork": false,
> > >     "url": "https://api.github.com/repos/octo-org/octo-repo",
> > >     "forks_url": "https://api.github.com/repos/octo-org/octo-repo/forks",
> > >     "keys_url": "https://api.github.com/repos/octo-org/octo-repo/keys{/key_id}",
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > here you can see github weebhook payload also using url, html_url for different context. I don't know what would be right url format though it's now resolved the null exception.
> > 
> > 
> > So `url` means APIURL, it's not the same as `HTMLURL`.
> 
> I build again using APIURL `URL: repo.APIURL(),` that is also working with jenkins plugin. so do I need to change it to repo.APIURL() ?

Yes, please.

**Review by CaiCandong — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**